quint/src/quintVerifier.ts

```diff
@@ -242,22 +191,33 @@ async function loadProtoDefViaReflection(): Promise<VerifyResult<ProtoPackageDef
   const serverReflectionService = reflectionProtoDescriptor.grpc.reflection.v1alpha.ServerReflection
   const reflectionClient = new serverReflectionService(APALACHE_SERVER_URI, grpc.credentials.createInsecure())
 
-  // Query reflection endpoint (retry if server is unreachable)
-  return retryWithTimeout(
-    () =>
-      new Promise<ServerReflectionResponse>((resolve, reject) => {
-        const call = reflectionClient.ServerReflectionInfo()
-        call.on('data', (r: ServerReflectionResponse) => {
-          call.end()
-          resolve(r)
-        })
-        call.on('error', (e: grpc.StatusObject) => reject(e))
+  // Wait for gRPC channel to come up, with 1sec pauses
+  if (retry) {
+    await (async () => {
+      for (;;) {
+        const grpcChannelState = reflectionClient.getChannel().getConnectivityState(true)
+        if (grpcChannelState == grpc.connectivityState.READY) {
```

Comment:
Great that this reliably ensures the service is ready! I did not have luck with this in the past, but maybe the timing taken by having to fetch the schema allows time to ensure the service if fully functional?

Reply:
This still usually loops ~3 times on my machine while the server is coming up.
I previously was doing this in a busy waiting loop, and that did not work. I suspect that there is a race with some async parts of the gRPC lib, which need runtime cycles to bring up the connection (even if the server is fully up).
I will leave a comment here, to note that we actually need the `setTimeout` to yield to to runtime at this point.

Reply:
See e78648f1

Reply:
But doesn't this mean we actually still have a potential race condition here? I'd rather keep using `ping` then trust our timeout will always save us.

Reply:
I cannot see a race here.

> I'd rather keep using ping then trust our timeout will always save us.

If the gRPC channel doesn't come up, ping will also fail. What do we gain by adding ping on top?